src/dmd/vsoptions.d

```diff
@@ -236,18 +242,27 @@ private:
     void detectVSInstallDir()
     {
         if (VSInstallDir is null)
-            VSInstallDir = getenv("VSINSTALLDIR");
+            VSInstallDir = getenv("VSINSTALLDIR"w);
 
         if (VSInstallDir is null)
             VSInstallDir = detectVSInstallDirViaCOM();
 
         if (VSInstallDir is null)
-            VSInstallDir = GetRegistryString(r"Microsoft\VisualStudio\SxS\VS7", "15.0"); // VS2017
+            VSInstallDir = GetRegistryString(r"Microsoft\VisualStudio\SxS\VS7", "15.0"w); // VS2017
+
+        if (VSInstallDir is null)
+        {
+            wchar[260] buffer = void;
+            // VS Build Tools 2017 (default installation path)
```

Comment:
See the commit msg. I don't remember precisely, been too long, and don't have any Build Tools installation available.

Reply:
Visual D detects the Build Tools by iterating over the keys in HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall, see https://github.com/dlang/visuald/blob/master/nsis/visuald.nsi#L1642

Reply:
Heh so there *is* a way... IIRC, I grepped the entire registry for some trace to the install dir but couldn't find anything.

I think adding that would be overkill / I'm not looking forward to implementing it ;). I'd hope that it can be found via COM nowadays; vswhere.exe has been bundled since VS 2017 v15.2 (but not with v15.0 and v15.1), so I don't know whether the COM interface was only introduced in v15.2 too.